src/cmd/backup-restore-tachyon.js

```diff
@@ -152,4 +166,66 @@ module.exports = class BackupRestoreTachyonCommand extends CLICommandBase {
 		}
 	}
 
+	async _getFilePathToRestore({ forceCloud, filePath, deviceId, inputDir }) {
+		const defaultFileName = path.join(inputDir, `manufacturing_backup_${deviceId}.zip`);
+		const defaultFileExists = await fileExists(defaultFileName);
+		if (filePath) {
+			const exists = await fileExists(filePath);
+			console.log(exists);
+			if (await fileExists(filePath)) {
+				this.ui.stdout.write(`Using zip file from ${filePath} ${os.EOL}`);
+				return filePath;
+			}
+			this.ui.stdout.write(`Unable to find file at ${filePath} ${os.EOL}`);
+			throw new Error('Unable to find file at ' + filePath);
+		}
+		if (forceCloud || !defaultFileExists) {
+			this.ui.stdout.write(`Downloading file at ${defaultFileName}${os.EOL}`);
+			const resp = await this.api.downloadManufacturingBackup({ deviceId });
+			const buffer = Buffer.from(resp);
+			await fs.writeFile(defaultFileName, buffer);
+		}
+		return defaultFileName;
+	}
+
+	async extractZipFile(filePath) {
+		const tmpOutputDir = await temp.mkdir('tachyon_restore');
+
+		let directory;
+		try {
+			// Try opening the zip — throws if not a zip
+			directory = await unzip.Open.file(filePath);
+		} catch (err) {
+			throw new Error(`Invalid zip file: ${filePath}`); // in case there is no zip or so throw an error
+		}
+		const roots = new Set(directory.files.map(f => f.path.split('/')[0]));
+
+		// Extract everything into tmpOutputDir
+		await fs.createReadStream(filePath)
+			// eslint-disable-next-line new-cap
+			.pipe(unzip.Extract({ path: tmpOutputDir }))
+			.promise();
+
+		// Decide what to return
+		if (roots.size === 1) {
+			const [root] = Array.from(roots);
+			const resolved = path.join(tmpOutputDir, root);
+			return resolved;
+		} else {
+			return tmpOutputDir;
+		}
```

Comment:
What does this do?

Reply:
When I uncompress on my mac the zip file that uncompressed into a dir hehe. So I was preparing this in case the files are into a dir. After that I recall Mac always uncompress the dir and doesn't mean the zip actually contains a folder inside. I'll remove that part to just return the tmpOutputDir hehe.